service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -128,17 +118,28 @@ public void setAdditionalResources(final List<Handler<Router>> resourceProviders
         registerAdditionalResources();
 
         final Future<Void> result = Future.future();
-        CompositeFuture.all(bindSecureHttpServer(), bindInsecureHttpServer())
-                .compose(r -> result.complete(), result);
+            bindSecureHttpServer().setHandler(res -> {
+                if (res.failed()){
+                    bindInsecureHttpServer().compose(r -> result.complete(), result);
+                } else if ( (config.isInsecurePortEnabled() && config.getInsecurePort(DEFAULT_PORT) != server.actualPort() )
+                        || (config.getInsecurePort() != Constants.PORT_UNCONFIGURED && config.getInsecurePort() != server.actualPort())) {
+                    bindInsecureHttpServer().compose(r -> result.complete(), result);
+                } else {
+                    result.complete();
+                }
```

Comment:
@jbtrystram The `isInsecurePortEnabled` javadoc states that `If {@code false}, it is guaranteed by the server that no opened port is insecure`.
With the above code and the conditions in `bindInsecureHttpServer()` however, an insecure port may indeed be opened even if `isInsecurePortEnabled == false`.

It's also what I would expect, that for example with a config like
````
insecurePortEnabled: false
insecurePort: 8088
````
_no_ insecure port 8088 would be opened.

Reply:
@calohmn The health check server does not consider this property at all. See https://www.eclipse.org/hono/docs/admin-guide/monitoring-tracing-config/#health-check-server-configuration

Reply:
This makes the usage of `isInsecurePortEnabled()` here a bug then, I suppose, and all the more confusing anyway, IMHO.

And I think it would be good to add a note in the above code then, that `InsecurePortEnabled` is always implicitly `true` for the health check. Otherwise someone used to how the insecure port is handled in other places of Hono might wonder why the flag is not considered here.

Reply:
@calohmn this code has already been changed on master. You are reviewing *old* code ...

Reply:
oh, sorry, happened when using 2 local Hono repos with one not being up-to-date.